webapp/src/containers/Rhs/index.tsx

```diff
@@ -1,24 +1,49 @@
-import React from 'react';
+import React, {useEffect} from 'react';
 
 import usePluginApi from 'hooks/usePluginApi';
 import {getprojectDetailsState, getRhsState} from 'selectors';
 
+import LinearLoader from 'components/loader/linear';
+
+import plugin_constants from 'plugin_constants';
+
+import AccountNotLinked from './accountNotLinked';
 import ProjectList from './projectList';
 import ProjectDetails from './projectDetails';
 
 const Rhs = (): JSX.Element => {
     const usePlugin = usePluginApi();
 
+    // Fetch the connected account details when RHS is opened
+    useEffect(() => {
+        if (getRhsState(usePlugin.state).isSidebarOpen) {
+            usePlugin.makeApiRequest(plugin_constants.pluginApiServiceConfigs.getUserDetails.apiServiceName);
+        }
+    }, []);
+
     if (!getRhsState(usePlugin.state).isSidebarOpen) {
         return <></>;
     }
 
     return (
-        <div className='height-rhs bg-sidebar padding-25'>
+        <div className='overflow-auto height-rhs bg-sidebar padding-25'>
+            {
+                usePlugin.getApiState(plugin_constants.pluginApiServiceConfigs.getUserDetails.apiServiceName).isLoading &&
+                <LinearLoader/>
+            }
+            {
+                usePlugin.getApiState(plugin_constants.pluginApiServiceConfigs.getUserDetails.apiServiceName).isError &&
+                <AccountNotLinked/>
+            }
             {
-                getprojectDetailsState(usePlugin.state).projectID ?
-                    <ProjectDetails title={getprojectDetailsState(usePlugin.state).projectName}/> :
-                    <ProjectList/>
+                !usePlugin.getApiState(plugin_constants.pluginApiServiceConfigs.getUserDetails.apiServiceName).isLoading &&
```

Comment:
Rather than writing this whole expression again and again, we can define a function something like below

 const getUserDetailsState = () => {
 
        const {isLoading, isError, isSuccess, data, error} = usePlugin.getApiState(plugin_constants.pluginApiServiceConfigs.getUserDetails.apiServiceName);
         return {isLoading, isError, isSuccess, data, error};
   }

Reply:
And then we can shorten all the expressions using something like - `getUserDetailsState().isLoading`

Reply:
yes we can also extract like this
```
    const {isLoading, isError, isSuccess, data} = usePlugin.getApiState(plugin_constants.pluginApiServiceConfigs.getUserDetails.apiServiceName)
```

Reply:
Yes, but most of the time we need to format `data` and `error` something like below
```
return {isLoading, isSuccess, isError, data: (data as CreateAndDeleteFeedResponse)?.message, error: ((error as FetchBaseQueryError)?.data as {error?: string})?.error};

```

Reply:
we can sync on this as in Azure we don't require this